components/RuiExpandCollapse/src/RuiExpandCollapse.ts

```diff
@@ -0,0 +1,223 @@
+/**
+ * Copyright Deloitte Digital 2019
+ *
+ * This source code is licensed under the BSD-3-Clause license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import {LitElement, html, property, CSSResultArray, TemplateResult} from 'lit-element';
+import {variables, layout} from './RuiExpandCollapse.css'
+
+// Update to include any possible type a value
+// can take, currenlty only have boolean open property
+type RuiExpandCollapsePropertyType = boolean;
+
+/**
+ * RuiExpandCollapse
+ */
+export class RuiExpandCollapse extends LitElement { 
+  /**
+   * Open property deals with the internal open/close 
+   * state. Mirrors the open attribute on the root element
+   */
+  @property({
+    type : Boolean,
+    reflect: true, // reflect attribute on parent element when internal state updates
+    converter: (value): boolean => value !== undefined, // check presence of open attribute
+  })
+  public get open(): boolean {
+    return this._open;
+  }
+
+  public set open(isOpen: boolean) {
+    const oldVal = this.open;
+    this._open = isOpen;
+    this.requestUpdate('open', oldVal);
+  }
+
+  /**
+   * onExpandCollapse is the handler function that is called
+   * when the user triggers an expand/collapse. This
+   * function should be overriden when trying to control
+   * the component externally
+   */
+  @property()
+  public onExpandCollapse = ():void => {
+    this.open = !this.open;
+  };
+
+  /**
+   * Internal open state of component
+   */
+  private _open: boolean = false;
+  private _collapseableEl: HTMLDivElement | null = null;
+  private _detailsSlotEl: HTMLSlotElement | null = null;
+  private _uuid: string = this._generateUUIDv4();
+
+  /**
+   * 
+   * The styles for the expand collapse
+   * @remarks
+   * If you are extending this class you can extend the base styles with super. Eg `return [super(), myCustomStyles]`
+   */
+  public static get styles(): CSSResultArray {
+    return [variables, layout];
+  }
+
+  private _generateUUIDv4(): string {
```

Comment:
Can we add a `TODO: To move this to core`

Reply:
@jennasalau fixed